bpf/lib/bpf_cgroup.h

```diff
@@ -391,4 +391,68 @@ __init_cgrp_tracking_val_heap(struct cgroup *cgrp, cgroup_state state)
 	return heap;
 }
 
+/**
+ * __set_task_cgrpid_tracker() Sets the cgrpid_tracker of a task.
+ * It reads tetragon_conf if not available then exit.
+ * If tetragon_conf is available then checks if the task
+ * execve_map_value->cgrpid_tracker is set, if so do nothing.
+ * If not set then fetch task cgroup and cgroup level, compare it against
+ * tetragon_conf->tg_cgrp_level which is the tracking cgroup level and set
+ * the task execve_map_value->cgrpid_tracker accordingly.
+ */
+static inline __attribute__((always_inline)) int
+__set_task_cgrpid_tracker(struct tetragon_conf *conf, struct task_struct *task,
+			  struct execve_map_value *execve_val, __u32 *error_flags)
+{
+	u32 flags = 0;
+	int subsys_idx = 0;
+	struct cgroup *cgrp;
+	__u64 cgrpid_tracker = 0;
+
+	/* Ensure that execve_val is not null */
+	if (unlikely(!conf) || unlikely(!execve_val))
+		return 0;
+
+	probe_read(&flags, sizeof(flags), _(&task->flags));
+	/* Skip kernel threads */
+	if (flags & PF_KTHREAD)
+		return 0;
+
+	/* Set the tracking cgroup id only if it was not set,
+	 * this avoids cgroup thread granularity mess!
+	 */
+	if (execve_val->cgrpid_tracker != 0)
+		return 0;
+
+	if (conf->tg_cgrp_subsys_idx != 0)
+		subsys_idx = conf->tg_cgrp_subsys_idx;
```

Comment:
I don't see why we need the branch here. If conf->tg_cgrp_subsys_idx is 0, then `subsys_idx` will end up being zero.


Can't we just do?
```
 get_task_cgroup(task, conf->tg_cgrp_subsys_idx, error_flags);
```

Reply:
Ah yes left over that indicated this is a cgroupv1 for readability, removed thanks